
Really don't enable any RPATHs if --disable-rpath is given.

--- php-5.0.4/acinclude.m4.norpath
+++ php-5.0.4/acinclude.m4
@@ -833,6 +833,7 @@
 dnl internal, don't use
 AC_DEFUN([_PHP_ADD_LIBPATH_GLOBAL],[
   PHP_RUN_ONCE(LIBPATH, $1, [
+    test "x$PHP_RPATH" != "xno" &&
     test -n "$ld_runpath_switch" && LDFLAGS="$LDFLAGS $ld_runpath_switch$1"
     LDFLAGS="$LDFLAGS -L$1"
     PHP_RPATHS="$PHP_RPATHS $1"
@@ -852,7 +853,11 @@
       _PHP_ADD_LIBPATH_GLOBAL([$ai_p])
     ],[
       if test "$ext_shared" = "yes"; then
+        if test "x$PHP_RPATH" = "xno"; then
+          $2="-L$ai_p [$]$2"
+        else
         $2="$ld_runpath_switch$ai_p -L$ai_p [$]$2"
+        fi
       else
         _PHP_ADD_LIBPATH_GLOBAL([$ai_p])
       fi
